cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -302,17 +302,24 @@ def process_snapd_timer():
     # Get the current snapd refresh timer; we know layer-snap has set this
     # when the 'snap.refresh.set' flag is present.
     timer = snap.get(snapname='core', key='refresh.timer').decode('utf-8')
+    if not timer:
+        # A subordinate wiped out our value, so we need to force it to be set
+        # again. Luckily, the subordinate should only wipe it out once, on
+        # first install, so this should remain stable afterward.
+        snap.set_refresh_timer(hookenv.config('snapd_refresh'))
+        timer = snap.get(snapname='core', key='refresh.timer').decode('utf-8')
 
     # The first time through, data_changed will be true. Subsequent calls
     # should only update leader data if something changed.
-    if data_changed('worker_snapd_refresh', timer):
+    if timer and data_changed('worker_snapd_refresh', timer):
```

Comment:
I think we should pull the `timer and` out of this condition.  It's possible an operators wants the timer to go back to the default, which is an empty string.  As is, we won't fall into the `leader_set({'snapd_refresh': timer})` below, so i *think* non-leaders will never get notified to set an empty refresh timer.